src/+groundwater/calculateGroundwaterRecharge.m

```diff
@@ -0,0 +1,131 @@
+    %}
+
+    % Load model settings
+    ModelSettings = io.getModelSettings();
+    NN = ModelSettings.NN; % Number of nodes
+    NL = ModelSettings.NL; % Number of layers
+
+    % Start Recharge calculations
+    if GroundwaterSettings.GroundwaterCoupling == 1 % Groundwater coupling is enabled
+        % Load needed variables
+        soilThick = GroundwaterSettings.soilThick; % cumulative soil layer thickness (from top to bottom)
+        indxAqLay = GroundwaterSettings.indxAqLay; % index of MODFLOW aquifer layers for each STEMMUS soil layer
+        aqLayers = GroundwaterSettings.aqLayers; % elevation of top surface level and all bottom levels of aquifer layers
+
+        % (a) Define the upper and lower boundaries of the moving balancing domain
+        % the moving balancing domain is located between depToGWT_strt and depToGWT_end
+        [depToGWT, indxGWLay] = groundwater.findPhreaticSurface(SoilVariables, KT, GroundwaterSettings);
+        depToGWT_end = depToGWT;
+        indxGWLay_end = indxGWLay;
+        % indxRchrg and indxRchrgMax are the indecies of the upper and lower levels of the moving boundary
+        % Following the HYDRUS-MODFLOW paper and also STEMMUS-MODFLOW, indxRchrg and indxRchrgMax are defined as in the next two lines
+        indxRchrgMax = max(indxGWLay_strt, indxGWLay_end) + 2; % the positive 2 is a user-specified value to define lower boundary of the moving boundary
+        % indxRchrg = min(indxGWLay_strt, indxGWLay_end) - 3; % the negative 2 or 3 is a user-specified value to define upper boundary of the moving boundary
+        % However, I comment the line above and use a slightly different way (ignore the moving layer boundaries and extract recharge from the two layers above groundwater table)
+        indxRchrg = min(indxGWLay_strt, indxGWLay_end) - 1;
+        indxRchrg_above = min(indxGWLay_strt, indxGWLay_end) - 2;
+
+        % (b) Call the fluxes to get the initial recharge
+        QL_h = EnergyVariables.QL_h; % liquid flux due to matric potential gradient
+        QL_T = EnergyVariables.QL_T; % liquid flux due to temperature gradient
+        QL_a = EnergyVariables.QL_a; % liquid flux due to air pressure gradient
+        QVH = EnergyVariables.QVH; % vapor water flux due to matric potential gradient
+        QVT = EnergyVariables.QVT; % vapor water flux due to temperature gradient
+        QVa = EnergyVariables.QVa; % vapor water flux due to air pressure gradient
+        Qtot = EnergyVariables.Qtot; % total flux (liquid + vapor)
+
+        % flip the fluxes (because STEMMUS calculations are from bottom to top, and MODFLOW needs the recharge from top to bottom)
+        QLh_flip = flip(QL_h(1, :));
+        QLT_flip = flip(QL_T(1, :));
+        QLa_flip = flip(QL_a(1, :));
+        QVH_flip = flip(QVH(1, :));
+        QVT_flip = flip(QVT(1, :));
+        QVa_flip = flip(QVa(1, :));
+        Q_flip = flip(Qtot(1, :));
```

Comment:
```suggestion
        % (b) Call the fluxes to get the initial recharge
        % flip the fluxes (because STEMMUS calculations are from bottom to top, and MODFLOW needs the recharge from top to bottom)
        QLh_flip = flip(EnergyVariables.QL_h(1, :)); % liquid flux due to matric potential gradient
        QLT_flip = flip(EnergyVariables.QL_T(1, :)); % liquid flux due to temperature gradient
        QLa_flip = flip(EnergyVariables.QL_a(1, :)); % liquid flux due to air pressure gradient
        QVH_flip = flip(EnergyVariables.QVH(1, :)); % vapor water flux due to matric potential gradient
        QVT_flip = flip(EnergyVariables.QVT(1, :)); % vapor water flux due to temperature gradient
        QVa_flip = flip(EnergyVariables.QVa(1, :)); % vapor water flux due to air pressure gradient
        Q_flip = flip(EnergyVariables.Qtot(1, :)); % total flux (liquid + vapor)
```